### ISSUE TITLE
In a npm/lerna/nx monorepo it can't find tsserver

### ISSUE THREAD

**Issue by faridsaid20** [pmizio/typescript-tools.nvim#71]
I tried https://github.com/pmizio/typescript-tools.nvim/pull/68
Did not help.

It is 100% reproducible and there is a workaround.
Prerequisite, install typescript in **root** (in my case **common_ui**) node_module only.
_common_ui/node_modules/typescript/_

### To trigger the bug:
Open monorepo and go to any js/ts/tsx/jsx file on any workspace which has its own node_module that does not have typescript:
_common_ui/packages/featureComponets/src/index.ts_

Result:
The error message :![image](https://github.com/pmizio/typescript-tools.nvim/assets/119894088/c8cf35ef-296e-4e4f-9990-6a7f669be9bf)

LspInfo:
![image](https://github.com/pmizio/typescript-tools.nvim/assets/119894088/b748fc5f-7f15-4f9a-8863-6e227c11205e)

If I run "npm root" from terminal from common_ui/packages/featureComponets/src dir:
![image](https://github.com/pmizio/typescript-tools.nvim/assets/119894088/a48db0cc-c71e-4a21-8fd6-a3b31bebac63)

### Workaround:
Make sure to open js/ts/tsx/jsx file in the root path first so that plugin finds node_module with typescript and then you are free to work on any file on the monorepo.

_root/.eslintrc.js_

Result:
LspInfo:
![image](https://github.com/pmizio/typescript-tools.nvim/assets/119894088/01256fe6-98e7-4fda-aca6-01834acb4d9f)

If I run "npm root" from terminal from common_ui dir:
![image](https://github.com/pmizio/typescript-tools.nvim/assets/119894088/a32d879b-1865-4b7f-9078-50f1547c6d1e)

**Comment by pmizio** [comment]
@faridsaid20 can you provide some public repo(it may be some library from npm) to tests? I need more informations there is come clear indication in subrepo about this workspace is indeed subrepo?

**Comment by faridsaid20** [comment]
> @faridsaid20 can you provide some public repo(it may be some library from npm) to tests? I need more informations there is some clear indication in subrepo about this workspace is indeed subrepo?

Sure,

Here is real world project that uses workspaces and typescript.
https://github.com/puppeteer/puppeteer/

I just tested this myself and it has the same behavior as on my project.

To trigger the bug.
Make sure to visit any js/ts file that is located under any package that have its own node_modules.
I tested in **_packages/browsers/src/CLI.ts_**

Work around is the same as I described.
Visit any js/ts file that is located in the root first.
